### PR TITLE
Gdx 560/fetch ruleset names and display on ruleset page

### PR DESCRIPTION
**Pull request by duckworthj39**
This pr kind of covers two jira tickets:
https://cddodatamarketplace.atlassian.net/browse/GDX-541
https://cddodatamarketplace.atlassian.net/browse/GDX-560

There are a number of ruleset selections here, including the old spectral ruleset and the crunch ruleset.

Think we probably need to consider some view specs or some specs using capybara now that we are getting the frontend behaviour solidified. Not something I want to include in this pr however.

### CODE REVIEW

File: app/lib/linters/spectral.rb [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+      stdout_str, stderr_str = system_command.capture3(
+        "npx spectral lint -f json #{path} --ruleset data/rulesets/#{ruleset_name}.yaml"
+      )
+      raise stderr_str if stderr_str.present?
       stdout_str
     rescue StandardError => e
       Rails.logger.error("npx spectral command failed: #{stderr_str}")
       Rails.logger.error(stdout_str)
-      raise SpectralLinterError, "npx spectral command failed: #{e.message}"
+      raise "npx spectral command failed: #{e.message}"
```
Comment (RobNicholsGDS) [inline app/lib/linters/spectral.rb:22]: [note]
I still think this "rescuing and then raising" pattern just obscures the error.

File: app/views/rulesets/index.html.erb [hunk 2]
```diff
@@ -11,8 +11,14 @@
             </h1>
           </legend>
           <div class="govuk-radios" data-module="govuk-radios">
+            <% Linters::Rulesets::ACTIVE_RULESETS.each do |ruleset_name| %>
+              <div class="govuk-radios__item">
+                <%= text_field_tag("ruleset_name", ruleset_name, class: "govuk-radios__input", type: "radio", name: "ruleset_name", checked: true) %>
```
Comment (RobNicholsGDS) [inline app/views/rulesets/index.html.erb:16]: [suggestion]

Is there a reason why `text_field_tag` is used here. Isn't [`radio_button_tag`](https://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html#method-i-radio_button_tag) more appropriate?

File: app/views/rulesets/index.html.erb [hunk 3]
```diff
@@ -11,8 +11,14 @@
...
+                <%= label_tag("gov-ruleset", t("rulesets.#{ruleset_name}.description"), class: "govuk-label govuk-radios__label", for: "gov-ruleset") %>
```
Comment (RobNicholsGDS) [inline app/views/rulesets/index.html.erb:17]: [change]

The [`for` field should point at the id of the radio button](https://stackoverflow.com/questions/1527721/using-label-for-on-radio-buttons) element the label applies to. Looking at the radio code, I think that will be `"ruleset_name_#{ruleset_name}"`. I can't see anything on the page that "gov-ruleset" will match.

File: app/views/rulesets/index.html.erb [hunk 4]
```diff
@@ -11,8 +11,14 @@
...
+              </div>
+            <% end %>
             <div class="govuk-radios__item">
-              <%= text_field_tag("ruleset_name", "government_ruleset", class: "govuk-radios__input", type: "radio", name: "ruleset_name", checked: true) %>
+              <%= text_field_tag("ruleset_name", "spectral", class: "govuk-radios__input", type: "radio", name: "ruleset_name", checked: true) %>
               <%= label_tag("gov-ruleset", "UK government ruleset", class: "govuk-label govuk-radios__label", for: "gov-ruleset") %>
```
Comment (RobNicholsGDS) [inline app/views/rulesets/index.html.erb:22]: [change]

I think the `for` here will be `"ruleset_name_spectral"`.

**Review by RobNicholsGDS — COMMENTED** [review]
Looks good. Just a couple of very minor changes needed.

**Comment by RobNicholsGDS** [comment]
For clarity - it is just the correction of the "for" settings for the two labels that needs doing.

File: app/views/rulesets/index.html.erb (hunk 2 above)
Comment (duckworthj39) [reply app/views/rulesets/index.html.erb:16]: The radio button tag doesn't work with the gov component, or at least I couldn't get it to work.

Comment (RobNicholsGDS) [reply app/views/rulesets/index.html.erb:16]: OK - though that seems strange.

Leave it as it is.

File: app/views/rulesets/index.html.erb (hunk 3 above)
Comment (DarrylMultisided) [reply app/views/rulesets/index.html.erb]: This is now done as per Rob's suggestion.

Comment (RobNicholsGDS) [reply app/views/rulesets/index.html.erb]: To clarify. Line 17 should be I think:
```ruby
<%= label_tag("gov-ruleset", t("rulesets.#{ruleset_name}.description"), class: "govuk-label govuk-radios__label", for: "ruleset_name_#{ruleset_name}") %>
```
Though worth firing the app up and checking the id of the radio button input to make sure I've constructed it correcly

File: app/views/rulesets/index.html.erb (hunk 4 above)
Comment (DarrylMultisided) [reply app/views/rulesets/index.html.erb]: I've since removed this bit of code - the 3 dynamic rulesets replace the hardcoded government ruleset.

Comment (RobNicholsGDS) [reply app/views/rulesets/index.html.erb]: I believe line 22 should be:
```ruby
 <%= label_tag("gov-ruleset", "UK government ruleset", class: "govuk-label govuk-radios__label", for: "ruleset_name_spectral") %>
```
But again, fire up the page and check the id there.

**Review by RobNicholsGDS — APPROVED** [review]
LGTM

**Pull request merged**